### ISSUE TITLE
pip install differences between platforms

### ISSUE THREAD

**Issue by cyrush** [LLNL/conduit#939]
To create the conduit python library, we build several compiled shared libs, and place those in a directory hierarchy with python source that imports them.

This is done with a combo of cmake and then distuils for the pure python parts. 

To support pip install, we provide a setup.py that drives cmake to build and install the module. 

While the module gets built and installed by cmake to a temp dir for the pip process and looks fine --  in some cases it gets mangled during the final install.

The pure python parts get siphoned into a different egg, breaking the directory structure that makes the module sound. 

I thought this might be a `zip_safe` issue, so I played with various settings for that. 

Here is an example of a broken install:
```
linux (rztopaz) idesign virtual env
Python 3.8.2
pip 22.0.4
setuptools-60.9.3
```

setup.py install creates:

```
conduit (includes compiled modules)
conduit-0.0.0-py3.8.egg (includes pure python module source, __init__.py, etc)
conduit-0.8.0-py3.9.egg-info
```

Here is an example of a working install:

```
macos
Python 3.8.11
pip 21.3.1 
setuptools-57.4.0
```

setup.py install creates:

```
conduit (includes compiled modules and pure python module sources files, __init__.py, etc)
conduit-0.8.0-py3.9.egg-info
```

These are just two cases of pip, not sure how scary the full landscape is ...

**Comment by cyrush** [comment]
For the broken case, `conduit-0.0.0-py3.8.egg`, will include a `conduit` directory that has the pure python files ....

(I can also get this file to be called   `conduit-0.8.0-py3.8.egg` by adding a version arg to the  `py_src/setup.py` but that doens't resolve the problem)

I also noticed that RPATHs aren't updated, I may need to build a flattened structured for pip to be able to repackage it.

**Comment by cyrush** [comment]
the strangeness on the linux system is not related to the top level `setup.py`, it starts with the `py_src/setup.py`

**Comment by cyrush** [comment]
env setting `SETUPTOOLS_USE_DISTUTILS=stdlib` seems to fix the issue on the linux system (stops cracked eggs from being created)

**Comment by cyrush** [comment]
resolved in #937
